### PR TITLE
config-tools: add validate_scenario_schema

### PR DESCRIPTION
**Pull request by yuchuyang**
add validate_scenario_schema to validate_scenario_setting and update the
excption handling.

Tracked-On: #5672
Signed-off-by: Yang,Yu-chu <yu-chu.yang@intel.com>

### CODE REVIEW

**Review by junjiemao1 — COMMENTED** [review]
Fine with me. @szhen11 Please help check if the formatted error messages fit your needs. Thanks.

File: misc/config_tools/scenario_config/scenario_cfg_gen.py [hunk 1]
```diff
@@ -91,35 +91,43 @@ def get_scenario_item_values(board_info, scenario_info):
...
+    xsd_doc.xinclude()
+    my_schema = xmlschema.XMLSchema11(etree.tostring(xsd_doc, encoding="unicode"))
 
-        # XMLSchema does not process XInclude. Use lxml to expand the schema which is feed to XMLSchema as a string.
-        xsd_doc = SCENARIO_SCHEMA_FILE
-        xsd_doc.xinclude()
-        my_schema = xmlschema.XMLSchema11(etree.tostring(xsd_doc, encoding="unicode"))
+    it = my_schema.iter_errors(scenario_info)
+    for idx, validation_error in enumerate(it, start=1):
+        key = "idx:path={},{}".format(idx, validation_error.path)
```
Comment (szhen11) [inline misc/config_tools/scenario_config/scenario_cfg_gen.py:115]: @yuchuyang Can we make the format of the key in ERROR_LIST same as the legacy format as below?
		{"vm:id=0,name": "VM name length should be in range [1,32] bytes"}
		{"vm:id=0,communication_vuart:id=1,target_vm_id": "target vm 2 disables legacy vuart 1 and all communication vuarts"}

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

File: misc/config_tools/scenario_config/scenario_cfg_gen.py (hunk 1 above)
Comment (yuchuyang) [reply misc/config_tools/scenario_config/scenario_cfg_gen.py]: @szhen11  updated.
Just keep in mind, this "target vm 2 disables legacy vuart 1 and all communication vuarts" haven't been implemented yet.

**Comment by wenlingz** [comment]
@terryzouhao please help approve this PR, thanks

**Review by terryzouhao — APPROVED** [review]

**Pull request merged**